api/reporting/queries/live/restructured_results/latest_workspaces_structured.sql

```diff
@@ -0,0 +1,38 @@
+-- latest workspaces, broken into logical structs for easier comprehension.
+SELECT
+    STRUCT(w.workspace_id,
+           w.name,
+           w.creator_id,
+           w.cdr_version_id,
+           w.creation_time) AS metadata,
+    STRUCT(w.creation_time,
+           w.last_accessed_time,
+           w.last_modified_time) AS history,
+    STRUCT( w.rp_additional_notes AS additional_notes,
+            w.rp_ancestry AS ancestry,
+            w.rp_anticipated_findings AS anticipated_findings,
+            w.rp_approved AS approved,
+            w.rp_commercial_purpose AS commercial_purpose,
+            w.rp_control_set AS control_set,
+            w.rp_disease_focused_research AS disease_focused_research,
+            w.rp_disease_of_focus AS disease_of_focus,
+            w.disseminate_research_other,
+            w.rp_drug_development AS drug_development,
+            w.rp_educational AS educational,
+            w.rp_ethics AS ethics,
+            w.rp_intended_study AS intended_study,
+            w.rp_methods_development AS methods_development,
+            w.rp_other_population_details AS other_population_details,
+            w.rp_other_purpose AS other_purpose,
+            w.rp_other_purpose_details AS other_purpose_details,
+            w.rp_population_health AS population_health,
+            w.rp_reason_for_all_of_us AS reason_for_all_of_us,
+            w.rp_review_requested AS review_requested,
+            w.rp_scientific_approach AS scientific_approach,
+            w.rp_social_behavioral AS social_behavioral,
+            w.rp_time_requested AS time_requested ) AS research_purpose,
+    STRUCT(w.billing_status AS status,
+           w.billing_account_type AS account_type ) AS billing,
+    STRUCT(w.needs_rp_review_prompt) AS process_status
```

Comment:
This one feels like a bit too much of a straight rename – which IIRC you'd tried to avoid doing, to reduce schema drift / random name-mapping challenges.

Reply:
For those columns that come from our MySQL App DB (currently all of them), the schema-defined column names in this dataset are exactly the  same. This isn't a schema  contract but a pretty/convenient/useful  presentation.

Also, it's  not  renaming the var but exposing it as `process_status.needs_rp_rreview_prompt`.

Reply:
I refactored all the rp review columns (I think) into one struct under `research_purpose`:

```json
"research_purpose": {
      "additional_notes": null,
      "ancestry": false,
      "anticipated_findings": null,
      "commercial_purpose": false,
      "control_set": true,
      "disease_focused_research": false,
      "disease_of_focus": null,
      "disseminate_research_other": null,
      "drug_development": false,
      "educational": false,
      "ethics": false,
      "intended_study": "",
      "methods_development": false,
      "other_population_details": null,
      "other_purpose": false,
      "other_purpose_details": null,
      "population_health": false,
      "reason_for_all_of_us": null,
      "scientific_approach": null,
      "social_behavioral": false,
      "review_rerquest": {
        "rerquested": false,
        "time_requested": null,
        "approved": null,
        "needs_user_prompt": "0"
      }
    }
```

Reply:
Ah, I misunderstood the syntax at first glance. I thought 'needs_rp_review_prompt' was being renamed to 'process_status'.